### DISCUSSION TITLE
switch off server side rendering window

### DISCUSSION THREAD

**Discussion by bigfooted** [Kitware/trame#265] [Q&A] [answered]
Hi, is there a way to do off-screen rendering like in paraview, so I don't get a window on the server? I tried simply adding `renderWindow.SetOffScreenRendering(1)` but that didn't seem to do anything.

**Comment by jourdain** [comment]
You should be able. I'm guessing you are using the Graphic version of the window rather than OSMesa/EGL.

Are you using VTK or ParaView (view proxy)? Which version?
On what OS?

**Reply by bigfooted** [reply, marked as answer]
Thanks for you reply. I guess I placed it at the wrong location, this works for me now.

```
renderer = vtkRenderer()
renderWindow = vtkRenderWindow()
# off-screen rendering 
renderWindow.SetOffScreenRendering(1)
renderWindow.AddRenderer(renderer)
```

Are there performance improvements that I can get by e.g. recompiling vtk ?

**Reply by jourdain** [reply]
Ok great!

For the speed, it should not. Since VTK is leverage your GPU for the rendering, you should be good.

Do you have some performance issue?

In general VTK-osmesa is slow because it is CPU only but the graphic version (X) and EGL should perform the same. 
When you have a headless server, you can not use the graphic version and need to use EGL to access your GPU capabilities.

Hope that make sense,

Seb

PS: Since we've been interacting a couple time already, I'm curious on what you are building with trame. Not sure if you can share it publicly but I would love to hear about it.

**Reply by bigfooted** [reply]
Sure, in one sentence: SU2 is an open source CFD and optimization software suite and I am exploring the options for a GUI:
https://su2code.github.io/docs_v7/home/
https://github.com/su2code/SU2

I start small or else nothing will ever get finished, so my first goal is to be able to read and visualize a mesh and a result file, set up a problem (physics solver, material choice, boundary conditions, numerics) and save the configuration file. In the next step, I would like to run the solver from the GUI (either local or remote) and do real time visualization of convergence information and monitoring points.
